error/s2n_errno.h

```diff
@@ -330,6 +330,7 @@ typedef enum {
     S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT,
     S2N_ERR_INVALID_SERIALIZED_CONNECTION,
     S2N_ERR_TOO_MANY_CAS,
+    S2N_ERR_NULL_CONFIG_SIGNAL,
```

Comment:
This shouldn't ever happen, but if something goes very wrong with a connection and this error is somehow surfaced to the user, should the error type maybe be internal instead of usage?